weld/exprs.rs

```diff
@@ -443,7 +431,35 @@ pub fn merge_expr(builder: Expr<Type>, value: Expr<Type>) -> WeldResult<Expr<Typ
                     return err;
                 }
             }
-            // TODO other builders...
+            DictMerger(ref elem_ty1, ref elem_ty2, _) => {
+                if let Struct(ref v_ty) = value.ty {
+                    if elem_ty1.as_ref() != &v_ty[0] {
```

Comment:
Add a check here to make sure `v_ty.len() >= 2` to prevent a panic in case it isn't